applications/distributed/grpc/grpc_endoscopy_tool_tracking/README.md

```diff
@@ -0,0 +1,153 @@
+Number of messages: 663
+Min Latency Message No: 379
+Min end-to-end Latency (ms): 24.854
+Avg end-to-end Latency (ms): 27.1886
+Max Latency Message No: 142
+Max end-to-end Latency (ms): 28.003
+
+Path 3: replayer,visualizer_op
+Number of messages: 663
+Min Latency Message No: 372
+Min end-to-end Latency (ms): 30.966
+Avg end-to-end Latency (ms): 33.325
+Max Latency Message No: 397
+Max end-to-end Latency (ms): 35.479
+
+Number of source messages [format: source operator->transmitter name: number of messages]:
+incoming_responses->output: 683
+replayer->output: 683
+```
+
+
+### C++
+
+```bash
+# Start the gRPC Server
+./dev_container build_and_run grpc_endoscopy_tool_tracking --run_args cloud [--language cpp]
+
+# Start the gRPC Client
+./dev_container build_and_run grpc_endoscopy_tool_tracking --run_args edge [--language cpp]
+```
+
+
+## Dev Container
+
+To start the the Dev Container, run the following command from the root directory of Holohub:
+
+```bash
+./dev_container vscode
+```
+
+### VS Code Launch Profiles
+
+#### C++
+
+The following launch profiles are available:
+
+- **(compound) grpc_endoscopy_tool_tracking/cpp (cloud & edge)**: Launch both the gRPC server and the client.
+- **(gdb) grpc_endoscopy_tool_tracking/cpp (cloud)**: Launch the gRPC server.
+- **(gdb) grpc_endoscopy_tool_tracking/cpp (edge)**: Launch the gRPC client.
+
+
+## Limitations & Known Issues
+
+- The connection between the server and the client is controlled by `rpc_timeout`. If no data is received or sent within the configured time, it assumes the call has been completed and hangs up. The `rpc_timeout` value can be configured in the [endoscopy_tool_tracking.yaml](./cpp/endoscopy_tool_tracking.yaml) file with a default of 5 seconds. Increasing this value may help on a slow network.
+- The server can serve one request at any given time. Any subsequent call receives a `grpc::StatusCode::RESOURCE_EXHAUSTED` status.
+- When debugging using the compound profile, the server may not be ready to serve, resulting in errors with the client application. When this happens, open [tasks.json](../../../.vscode/tasks.json), find `Build grpc_endoscopy_tool_tracking (delay 3s)`, and adjust the `command` field with a higher sleep value.
+- The client is expected to exit with the following error. It is how the client application terminates when it completes streaming and displays the entire video.
+  ```bash
+  [error] [program.cpp:614] Event notification 2 for entity [video_in__outgoing_requests] with id [33] received in an unexpected state [Origin]
+  ```
+
+
+## Containerize the application
+
+To containerize the application using [Holoscan CLI](https://docs.nvidia.com/holoscan/sdk-user-guide/cli/cli.html), first build the application using `./dev_container build_and_install grpc_endoscopy_tool_tracking`, run the `package-app.sh` script in the [cpp](./cpp/package-app.sh) directory and then follow the generated output to package and run the application.
```

Comment:
NIT: since the README has not mentioned about installing holoscan package at all, a user running the generated command to package the app would encounter `No module named holoscan` error. A word on pre-requisite package will be helpful.

Also, the generated command can only work with HSDK >=2.7 due to the `--add` option.

Reply:
Thank you @MMelQin. I have made 2.7 as minimum required HSDK version